### PR TITLE
feat: run GHA build job locally with act

### PR DESCRIPTION
**Pull request by lmilbaum**
## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
I usually use Makefiles although left it out of this project due to the number of Windows users. We face this challenge:

- If we put anything in the Makefile which can't be done otherwise, we might accidentally have a project which is hostile to Windows contributors
- If everything in the Makefile maps to pnpm ... then we are adding this file just to save 5 characters for each command

So @lmilbaum is there any strong benefit from this?

**Comment by lmilbaum** [comment]
@rarkins Thank you for your feedback. I'm testing something with the PR. It is in draft.  Can we review its value or/and alternative options in a later stage?

File: .github/actions/setup-node/action.yml
Comment (viceice) [inline .github/actions/setup-node/action.yml:52]: why this is required?

Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: Setup node uses pnpm but the corepack is not enabled yet so the binary is missing.

Comment (viceice) [reply .github/actions/setup-node/action.yml:52]: then enable corepack instead

Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: It is installed with the node setup

Comment (viceice) [reply .github/actions/setup-node/action.yml:52]: sure, but corepack is also available if nodejs is already available

Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: nodejs is not available with the act images I tested

**Comment by lmilbaum** [comment]
@viceice ping

File: .github/actions/setup-node/action.yml
Comment (viceice) [reply .github/actions/setup-node/action.yml:52]: 😕, with this change we need a new regex manager to update the pnpm version

Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: Not sure it worth the efforts. This section will be removed as soon as this PR is merged:
https://github.com/actions/setup-node/pull/651

Comment (rarkins) [reply .github/actions/setup-node/action.yml:52]: Do we want to merge this before then? That PR is conflicted and waiting for someone to resume it and then hopefully the maintainers to merge it

Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: I do think it has value in merging it. Waiting for a general approval and then I'll rebase.

Comment (rarkins) [reply .github/actions/setup-node/action.yml:52]: We'll need a regex manager added to keep it updated though. It's already out of date :)

**Review by rarkins — CHANGES_REQUESTED** [review]
Add regex manager for updating pnpm version in action

File: .github/actions/setup-node/action.yml
Comment (lmilbaum) [reply .github/actions/setup-node/action.yml:52]: Done

Comment (viceice) [reply .github/actions/setup-node/action.yml:52]: I would prefer to leave the corepack enable when not on act. maybe we should build our own act image to not need the huge default act image.

File: .github/actions/setup-node/action.yml
Comment (viceice) [inline .github/actions/setup-node/action.yml:63]: this will probably not work when there is a pnpm in place

File: docs/development/local-development.md
Comment (viceice) [inline docs/development/local-development.md:87]: why do we need the large image?

File: .github/workflows/build.yml
Comment (lmilbaum) [reply .github/workflows/build.yml:474]: This step should not be enabled when running the pipeline with act

File: docs/development/local-development.md
Comment (lmilbaum) [reply docs/development/local-development.md:87]: I don't remember now why the large image is required. I do remember that the smaller image doesn't work.

**Comment by rarkins** [comment]
Can this be moved forward to merging or should it be closed?

**Comment by lmilbaum** [comment]
> Can this be moved forward to merging or should it be closed?

Thanks @rarkins for reminding me of this PR. Closing.

**Pull request closed without merging**
